Create a shared form component #27

ravighevariya10 opened the pull request:
Created a shared PSAForm

ravighevariya10 committed:
Created a shared PSAForm

Adamws33 commented:
Discussed needed changes with Ravi in a phone call

ravighevariya10 committed:
Made the component to set the fields dynamically

ravighevariya10 committed:
Merge branch 'develop' into PSAForm

ravighevariya10 committed:
installed fontsource dependency

Adamws33 commented on src/psaform/psaform.jsx:
```diff
@@ -0,0 +1,245 @@
+      </Grid>
+
+      {/* Dynamically Render Text Fields */}
+      {textFields.map((field, index) => (
```
How things are currently made we have textFields and Checkboxes (line 155) separate.  This means that if we have a form that needs a text field then a checkbox then a textfield we will not be able to create that.  

What i recommend is create a single prop called formContent and that should be an array of objects.  One of the properties of each object should be type.  This will also not limit us to text boxes and checkboxes, we can then create an endless amount of types as new forms are created.  Then we just have to add logic to handle the new types.  

let me know if you'd like a little more guidance in getting this done.

ravighevariya10 committed:
The PSAForm component has been updated to utilize a single fields prop that consolidates all different form fields into one cohesive structure.

ravighevariya10 committed:
Updated the comment

ravighevariya10 committed:
added the missing prop

ravighevariya10 committed:
Updated the prop types

ravighevariya10 committed:
updated props

YazidZidane commented on src/psaform/psaform.jsx:
```diff
@@ -0,0 +1,212 @@
+  fields, // Single prop for both text fields and checkboxes
+  buttons,
+  consentRedux,
+  pirschAnalytics,
```
remove `consentRedux` and `pirschAnalytics` and the `useEffect` in the code as well

ravighevariya10 committed:
Updated the proptypes

ravighevariya10 committed:
updated the props

YazidZidane commented on src/psaform/psaform.jsx:
```diff
@@ -0,0 +1,212 @@
+};
+
+PSAForm.propTypes = {
+  apiUrl: PropTypes.string.isRequired,
```
Can you give these props a short description in comment?

YazidZidane commented on src/psaform/psaform.jsx:
```diff
@@ -0,0 +1,212 @@
+    })
+  ).isRequired,
+  consentRedux: PropTypes.bool,
+  pirschAnalytics: PropTypes.func.isRequired,
```
Can you just remove the `isRequired` for all the props because not all of them should be strictly required.

YazidZidane commented on src/psaform/psaform.jsx:
```diff
@@ -0,0 +1,190 @@
+}) => {
+  const [isSubmitDisabled, setIsSubmitDisabled] = useState(true);
+
+  const remove = (arr, value) => arr.filter((item) => item !== value);
```
Can you put this function into `handleCheckboxChange()` since it's only being used there

ravighevariya10 committed:
1. Gave a short description about the props
2. Removed the consentRedux and pirschAnalytics and the useEffect in the code as well
3. Removed the isRequired from all the props
4. put remove function into the handleCheckboxChange()

ravighevariya10 committed:
Updated the initialFormData

ravighevariya10 committed:
Put log to check the formData structure

ravighevariya10 committed:
Fixed the form submission
